api/turing/config/config.go

```diff
@@ -333,14 +334,23 @@ type MLPConfig struct {
 	MLPEncryptionKey string `validate:"required"`
 }
 
+// OpenapiConfig contains the settings for the OpenAPI specs used for validation and Swagger UI
 type OpenapiConfig struct {
 	// ValidationEnabled specifies whether to use OpenAPI validation middleware,
 	// which validates HTTP requests against the spec.
 	ValidationEnabled bool
 	// SpecFile specifies the file path containing OpenAPI v3 spec
 	SpecFile string
-	// Optional. Defines a configuration to be used for serving Swagger UI as a single-page app
+	// Config file to be used for serving swagger.
 	SwaggerUIConfig *SinglePageApplicationConfig
+	// Where the merged spec file yaml should be served.
+	YAMLServingPath string
+	// Optional. Overrides the file before running the Swagger UI.
+	SpecOverrideFile *string
+}
+
+func (c *OpenapiConfig) SpecData() ([]byte, error) {
+	return utils.MergeTwoYamls(c.SpecFile, *c.SpecOverrideFile)
```

Comment:
Wouldn't this cause a nil pointer exception when you try to de-reference `nil` `SpecOverrideFile`?

Reply:
Yeah it would. Let me correct that. Thanks